lib/src/ast_extensions.dart

```diff
@@ -184,6 +188,12 @@ extension ExpressionExtensions on Expression {
       MethodInvocation(:var argumentList) =>
         argumentList.arguments.canSplit(argumentList.rightParenthesis),
 
+      // Note: Using a separate case instead of `||` for this type because
+      // Dart 3.0 reports an error that [argumentList] has a different type
```

Comment:
Is this fixed in a more recent Dart SDK? Can we bump the constraint instead of hacking around the older SDK?

Reply:
> Is this fixed in a more recent Dart SDK?

I assume so? I only noticed this because it failed the 3.0.0 CI when I uploaded the PR, but didn't fail the newer one.

> Can we bump the constraint instead of hacking around the older SDK?

I can, but the change here is so innocuous and messing with the CI config so annoying that I'm inclined to just leave this as-is.